sticht/rollbacks/sources/splunk.py

```diff
@@ -116,52 +115,51 @@ def process_result(self, result: Optional[List[Dict[Any, Any]]], earliest_timest
         number of results from a query against configured thresholds to determine
         whether or not to rollback or not
         """
-        if earliest_timestamp_seconds != 'now':
-            if earliest_timestamp_seconds < self.start_timestamp_seconds:
-                self.bad_before_mark = self.is_window_bad(result)
-        else:
+        if earliest_timestamp_seconds > self.start_timestamp_seconds:
             self.bad_after_mark = self.is_window_bad(result)
+        else:
+            self.bad_before_mark = self.is_window_bad(result)
 
         old_failing = self.failing
         self.failing = self.bad_after_mark and not self.bad_before_mark
+
         if self.failing == (not old_failing):
             self.on_failure_callback(self)
 
     def is_window_bad(self, result) -> bool:
+        # Results return an array | Number returns a number
         if self._query_type == 'results':
-            if self._lower_bound and len(result) < self._lower_bound:
-                return True
-            elif self._upper_bound and len(result) > self._upper_bound:
+            result_num = len(result)
+        elif self._query_type == 'number':
+            result_num = next(iter(result[0].values())) if len(result) else len(result)
+
+        if self._lower_bound and self._upper_bound:
+            if result_num > self._lower_bound and result_num < self._upper_bound:
```

Comment:
Minor typo, I think we want the opposite here, `if < lower_bound or > upper_bound`, or ! the entire thing, ie `not (> lower_bound AND < upper_bound)`. Same goes for the comparisons below for the single bound being declared

I wonder if there should be a test for is_window_bad that would've caught this?

Reply:
Yeah this was a bit confusing at first, but I think that logic is correct because we're explicitly looking for bad data. If we check for anything outside the bounds, that means the data is good.

We don't specifically test is_window_bad, but we do test if the data has been failing before deployment (bad_before_mark).
https://github.com/Yelp/sticht/blob/master/tests/rollbacks/sources/test_splunk.py#L38